model/riscv_insts_vext_total.sail

```diff
@@ -0,0 +1,9655 @@
+  result      : vector('n, dec, bits('m)) = undefined;
+  mask_helper : vector('n, dec, bool)     = undefined;
+
+  /* Value of vstart must be 0 */
+  status = assert_vstart(0);
+
+  /* If masking is enabled, then dest reg cannot be v0 */
+  if vm == 0b0 & vd == vreg_name("v0") then {
+    status = RETIRE_FAIL
+  };
+
+  /* Dest reg cannot be the same as source reg */
+  if vd == vs2 then {
+    status = RETIRE_FAIL
+  };
+
+  if vm_val      == undefined /*|
+     vs2_val     == undefined |
+     vd_val      == undefined |
+     mask_helper == undefined*/ then {
+    status = RETIRE_FAIL
+  };
+
+  if status == RETIRE_SUCCESS then {
+    (result, mask_helper) = init_masked_result(num_elem, vsew_bits, lmul, vd_val, vm_val);
+
+    var total : int = 0;
+    foreach (i from 0 to (num_elem - 1)) {
+      if mask_helper[i] == true then {
+	      result[i] = to_bits(vsew_bits, total);
+        if vs2_val[i] == true then total = total + 1
+      }
+    };
+    
+    write_vreg(num_elem, vsew_bits, lmul, vd, result)
+  };
+
+  if status == RETIRE_FAIL then handle_illegal();
+  status
+}
+
+mapping clause assembly = VIOTA_M(vm, vs2, vd)
+  <-> "viota.m" ^ spc() ^ vreg_name(vd) ^ sep() ^ vreg_name(vs2) ^ maybe_vmask(vm)
+
+
+/* **************************VID_V**************************************** */
+
+union clause ast = VID_V : (bits(1), regidx)
+
+mapping clause encdec = VID_V(vm, vd)
+  <-> 0b010100 @ vm @ 0b00000 @ 0b10001 @ 0b010 @ vd @ 0b1010111
+
+function clause execute(VID_V(vm, vd)) = {
+  status         : Retired = RETIRE_SUCCESS;
+  let vsew_bits  : int  = get_vtype_vsew();
+  let lmul       : real = get_vtype_LMUL();
+  let num_elem   : int  = get_num_elem(lmul, vsew_bits);
+
+  assert(8 <= vsew_bits & vsew_bits <= 64);
+  assert(0 <= num_elem & num_elem <= sizeof(vlen));
+  let 'n = num_elem;
+  let 'm = vsew_bits;
+  
+  let vm_val  : vector('n, dec, bool)     = read_vmask(num_elem, vm, vreg_name("v0"));
+  let vd_val  : vector('n, dec, bits('m)) = read_vreg(num_elem, vsew_bits, lmul, vd);
+  result      : vector('n, dec, bits('m)) = undefined;
+  mask_helper : vector('n, dec, bool)     = undefined;
+
+  if vm_val      == undefined /*|
+     vd_val      == undefined |
+     mask_helper == undefined*/ then {
+    status = RETIRE_FAIL
+  };
+
+  if status == RETIRE_SUCCESS then {
+    (result, mask_helper) = init_masked_result(num_elem, vsew_bits, lmul, vd_val, vm_val);
+
+    foreach (i from 0 to (num_elem - 1)) {
+      if mask_helper[i] == true then {
+        result[i] = to_bits(vsew_bits, i)
+      }
+    };
+    
+    write_vreg(num_elem, vsew_bits, lmul, vd, result)
+  };
+
+  if status == RETIRE_FAIL then handle_illegal();
+  status
+}
+
+mapping clause assembly = VID_V(vm, vd)
+  <-> "vid.v" ^ spc() ^ vreg_name(vd) ^ maybe_vmask(vm)
+
+
+/* **************************NITYPE(Narrowing)**************************************** */
+
+union clause ast = NITYPE : (nifunct6, bits(1), regidx, regidx, regidx)
+
+mapp
```

Comment:
This mixed assignment / definition using a tuple won't be supported by the next Sail version. The `result_real` line isn't necessary anyway and can be removed.  Instead put a `let` in front of the tuple to make it a definition rather than an assignment.

Reply:
That's strange, github has changed the line range to a different piece of code entirely.  Here's a patch instead:
```
--- a/model/riscv_insts_vext_total.sail
+++ b/model/riscv_insts_vext_total.sail
@@ -8285,9 +8285,8 @@ function process_rfvv_widen(funct6, vm, vs2, vs1, vd, num_elem, vsew_bits, lmul)
   vs1_0 : bits('double_vsew) = read_single_element(double_vsew, 0, double_lmul, vs1);
   foreach (i from 0 to (num_elem - 1)) {
     if mask_helper[i] == true then {
-      result_real : real = undefined;
       /* Currently Ordered/Unordered sum do the same operations */
-      (fptype, sign, result_real) = fpreal_add(fp_to_real(vs2_val[i]), fp_to_real(vs1_0));
+      let (fptype, sign, result_real) = fpreal_add(fp_to_real(vs2_val[i]), fp_to_real(vs1_0));
       vs1_0 = real_to_fp(result_real)
     }
   };
```